pkg/operator/controller/ingress/status.go

```diff
@@ -57,17 +43,43 @@ func (r *reconciler) syncIngressControllerStatus(ic *operatorv1.IngressControlle
 	return nil
 }
 
-// computeIngressStatusConditions computes the ingress controller's current state.
-func computeIngressStatusConditions(oldConditions []operatorv1.OperatorCondition, deployment *appsv1.Deployment) []operatorv1.OperatorCondition {
-	oldAvailableCondition := getIngressAvailableCondition(oldConditions)
+// updateConditions adds or updates matching conditions, and updates
+// the transition time if details of a condition have changed. Returns
+// the updated condition array.
+func updateConditions(conditions []operatorv1.OperatorCondition, updates ...operatorv1.OperatorCondition) []operatorv1.OperatorCondition {
```

Comment:
this function is bugged and doesn't merge properly, I'm in the middle of redoing it but it makes me wonder if there's a library out there somewhere to merge lists of conditions

Reply:
> this function is bugged and doesn't merge properly,

Yeah, you gotta actually update the status, reason, and message of `current` if it's not nil.

> I'm in the middle of redoing it but it makes me wonder if there's a library out there somewhere to merge lists of conditions

The hard part is doing it generically.  In recent Git history, we had both `setOperatorStatusCondition` and `setIngressStatusCondition` for this reason.  Also, there are all sorts of nuances (such as when to update the last transition time of conditions) that need to be just right (otherwise we might be able to use `SetOperatorCondition` in library-go).